src/protocol/CommandHeader.cpp

```diff
@@ -106,11 +104,10 @@ void SendMessageRequestHeader::SetDeclaredFieldOfCommandHeader(map<string, strin
   requestMap.insert(pair<string, string>("bornTimestamp", UtilAll::to_string(bornTimestamp)));
   requestMap.insert(pair<string, string>("flag", UtilAll::to_string(flag)));
   requestMap.insert(pair<string, string>("properties", properties));
-#ifdef ONS
   requestMap.insert(pair<string, string>("reconsumeTimes", UtilAll::to_string(reconsumeTimes)));
   requestMap.insert(pair<string, string>("unitMode", UtilAll::to_string(unitMode)));
-#endif
-  requestMap.insert(pair<string, string>("batch", UtilAll::to_string(batch)));

```

Comment:
here i think there will problem for send message to broker with signature, because produce signature by spas_sign in doBeforeRequest depends on all header fields

void ClientRPCHook::doBeforeRequest(const string& remoteAddr, RemotingCommand& request) {
  。。。。
  header->SetDeclaredFieldOfCommandHeader(requestMap);
  map<string, string>::iterator it = requestMap.begin();
  for (; it != requestMap.end(); ++it) {
    totalMsg.append(it->second);
  }
  if (request.getMsgBody().length() > 0) {
    totalMsg.append(request.getMsgBody());
  }
  char* pSignature =
      rocketmqSignature::spas_sign(totalMsg.c_str(), totalMsg.size(), sessionCredentials.getSecretKey().c_str());
}

Reply:
The server will not verify this field in the command header.